app/main/lib/shared_models/video_model.py

```diff
@@ -111,19 +111,20 @@ def search_by_context(self, context):
             raise e
 
     def search(self, task):
-        body, threshold, limit = media_crud.parse_task_search(task)
-        video, temporary = media_crud.get_object(body, Video)
-        if video.hash_value is None:
-            callback_url =  Presto.add_item_callback_url(app.config['ALEGRE_HOST'], "video")
-            if task.get("doc_id") is    None:
-                task["doc_id"] = str(uuid.uuid4())
-            response = json.loads(Presto.send_request(app.config['PRESTO_HOST'], "video__Model", callback_url, task, False).text)
-            # Warning: this is a blocking hold to wait until we get a response in 
-            # a redis key that we've received something from presto.
-            result = Presto.blocked_response(response, "video")
-            video.hash_value = result.get("body", {}).get("result", {}).get("hash_value")
-        if video:
-            if self.tmk_file_exists(video):
+        temporary = False
+        try:
+            body, threshold, limit = media_crud.parse_task_search(task)
+            video, temporary = media_crud.get_object(body, Video)
+            if video.hash_value is None:
+                callback_url =  Presto.add_item_callback_url(app.config['ALEGRE_HOST'], "video")
+                if task.get("doc_id") is    None:
+                    task["doc_id"] = str(uuid.uuid4())
+                response = json.loads(Presto.send_request(app.config['PRESTO_HOST'], "video__Model", callback_url, task, False).text)
+                # Warning: this is a blocking hold to wait until we get a response in 
+                # a redis key that we've received something from presto.
+                result = Presto.blocked_response(response, "video")
+                video.hash_value = result.get("body", {}).get("result", {}).get("hash_value")
+            if video:
```

Comment:
Technically we should be able to drop the `if video` conditional. We've already tried `video.hash_value` on line 118; so, if the video is None we'll have failed earlier than this line already. We must have a video at this point basically.

Reply:
agreed